### PR TITLE
MIOpen: Honor Max Dim

### PR DESCRIPTION
**Pull request by mwootton**
Don't use MIOpen for a tensor with too many dimensions.

### CODE REVIEW

**Comment by iotamudelta** [comment]
@mwootton something is off w/ the MIOPEN_MAX macro for all other targets...

**Comment by iotamudelta** [comment]
my concern is that `getenv()` is a slow system call. Can we not cache this?

**Comment by mwootton** [comment]
> my concern is that `getenv()` is a slow system call. Can we not cache this?

Yes, yes we can.

File: aten/src/ATen/native/Convolution.cpp [hunk 1]
```diff
@@ -3,6 +3,9 @@
 
 #include "ATen/Config.h"
 
+static int MIOPEN_DIM_MAX = 4;
+static bool MIOPEN_ENABLED = getenv("DISABLE_MIOPEN") != NULL;
```
Comment (iotamudelta) [inline aten/src/ATen/native/Convolution.cpp]: Can we mark them const?

**Comment by iotamudelta** [comment]
LGTM - with the caveat that we'll see what upstream has to say about `getenv()`

**Pull request merged**